navigation-material/src/main/java/com/google/accompanist/navigation/material/BottomSheetNavigator.kt

```diff
@@ -165,63 +174,71 @@ public class BottomSheetNavigator(
-            // these
-            entry.lifecycle.currentState.isAtLeast(Lifecycle.State.STARTED)
-        }
+    val sheetContent: @Composable ColumnScope.() -> Unit =
+        movableContentWithReceiverOf<ColumnScope> {
+            val saveableStateHolder = rememberSaveableStateHolder()
+            val transitionsInProgressEntries by transitionsInProgress.collectAsState()
 
-        // Mark all of the entries' transitions as complete, except for the entry we are
-        // currently displaying because it will have its transition completed when the sheet's
-        // animation has completed
-        DisposableEffect(backStackEntries) {
-            transitionsInProgressEntries.forEach {
-                if (it != latestEntry) state.markTransitionComplete(it)
+            // The latest back stack entry, retained until the sheet is completely hidden
+            // While the back stack is updated immediately, we might still be hiding the sheet, so
+            // we keep the entry around until the sheet is hidden
+            val retainedEntry by produceState<NavBackStackEntry?>(initialValue = null) {
+                backStack
+                    .transform { backStackEntries ->
+                        // Always hide the sheet when the back stack is updated
+                        // Regardless of whether we're popping or pushing, we always want to hide
+                        // the sheet first before deciding whether to re-show it or keep it hidden
+                        sheetState.hide()
+                        emit(backStackEntries.lastOrNull())
+                    }
+                    .collectLatest { value = it }
             }
-            onDispose { }
-        }
 
-        SheetContentHost(
-            columnHost = columnScope,
-            backStackEntry = latestEntry,
-            sheetState = sheetState,
-            saveableStateHolder = saveableStateHolder,
-            onSheetShown = { backStackEntry ->
-                state.markTransitionComplete(backStackEntry)
-            },
-            onSheetDismissed = { backStackEntry ->
-                // Sheet dismissal can be started through popBackStack in which case we have a
-                // transition that we'll want to complete
-                if (transitionsInProgressEntries.contains(backStackEntry)) {
-                    state.markTransitionComplete(backStackEntry)
-                } else {
-                    state.pop(popUpTo = backStackEntry, saveState = false)
+            if (retainedEntry != null) {
+                LaunchedEffect(retainedEntry) {
+                    sheetState.show()
                 }
             }
-        )
-    }
+
+            SheetContentHost(
+                backStackEntry = retainedEntry,
+                sheetState = sheetState,
+                saveableStateHolder = saveableStateHolder,
+                onSheetShown = {
+                    transitionsInProgressEntries.forEach(state::markTransitionComplete)
```

Comment:
How does this work when moving between multiple screens (i.e., from Sheet A then `navigate(SheetB)`, `navigate(SheetC)` - do we move directly to `SheetC` (in which case this makes sense - finishing the show of `SheetC` should mark both `SheetB` and `SheetC` as completed) or do we see `SheetB`, then subsequently see `SheetC`?

Reply:
If we're being consistent with our UX we'd probably want to first see `SheetB` and then `SheetC`. I'm going to leave addressing this for a follow-up PR, I believe this behavior is okay for now.